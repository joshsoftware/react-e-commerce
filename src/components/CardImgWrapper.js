import React from 'react';
import { CardImg } from 'reactstrap';
import { PropTypes } from 'prop-types';
const CardImgWrapper = ({ width, src, top, bottom, className, tag }) => {
  return (
    <CardImg
      top={top}
      bottom={bottom}
      tag={tag}
      width={width}
      className={className}
      src={src}
      alt={'Product Image'}
    />
  );
};
export default CardImgWrapper;

CardImgWrapper.propTypes = {
  width: PropTypes.string,
  src: PropTypes.string.isRequired,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  className: PropTypes.string,
  top: PropTypes.bool,
  bottom: PropTypes.bool
};
