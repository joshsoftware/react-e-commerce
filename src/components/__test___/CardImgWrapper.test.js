import React from 'react';
import { render } from '@testing-library/react';
import CardImgWrapper from '../CardImgWrapper';

describe('Card Image', () => {
  it('Must Return Card Image', () => {
    const { asFragment } = render(
      <CardImgWrapper
        src={
          'https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg'
        }
      />
    );
    //console.log(card_image)
    expect(asFragment()).toMatchSnapshot();
  });
});
