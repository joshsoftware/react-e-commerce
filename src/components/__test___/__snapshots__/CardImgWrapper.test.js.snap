// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Image Must Return Card Image 1`] = `
<DocumentFragment>
  <img
    alt="Product Image"
    class="card-img"
    src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
  />
</DocumentFragment>
`;
